### PR TITLE
Adding tests for PrefixContainer

### LINKED ISSUE #4045: Add unit tests for PrefixContainer
As part of #4093 I'm adding tests for `ContainsPrefix` but there's a fair bit of complex functionality on this class that doesn't have any unit tests.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #4045

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/PrefixContainerTest.cs [hunk 1]
```diff
@@ -37,13 +37,13 @@ public void ContainsPrefix_HasEntries_EmptyString_True()
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/PrefixContainerTest.cs]: What about `GetKeysFromPrefix`

**Comment by pranavkm** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/DictionaryModelBinder.cs [hunk 2]
```diff
@@ -56,7 +56,7 @@ public class DictionaryModelBinder<TKey, TValue> : CollectionModelBinder<KeyValu
 
             // Attempt to bind dictionary from a set of prefix[key]=value entries. Get the short and long keys first.
             var keys = enumerableValueProvider.GetKeysFromPrefix(bindingContext.ModelName);
-            if (!keys.Any())
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/DictionaryModelBinder.cs]: :heart_eyes_cat:

**Comment by rynowak** [comment]
:shipit:

**Pull request closed without merging**